src/cdp/webSocketTransport.ts

```diff
@@ -28,13 +28,21 @@ export class WebSocketTransport implements ITransport {
     const isSecure = !url.startsWith('ws://');
     const targetAddressIsLoopback = await isLoopback(url);
 
-    const ws = new WebSocket(url, [], {
-      headers: { host: 'localhost' },
+    const options = {
+      headers: {},
       perMessageDeflate: false,
       maxPayload: 256 * 1024 * 1024, // 256Mb
       rejectUnauthorized: !(isSecure && targetAddressIsLoopback),
       followRedirects: true,
-    });
+    };
+
+    if (targetAddressIsLoopback) {
+      options.headers = {
+        host: 'localhost',
+      };
+    }
```

Comment:
This is intentionally added all the time -- by default chrome will not respond to connections made remotely (i.e. to a non-local Host) as a security measure against attacks from the browser.

Reply:
in private network, websocket may proxied by NGINX or other reverse proxy.

change it to add `localhost` header only when url hostname is ip.

Reply:
I think I'd rather keep doing it all the time.

Setting the host to localhost _could_ break in the reverse proxy scenario and require changes to the proxy to allow it to work.

Not doing so _will_ break lookups that use DNS and don't use a reverse proxy to correct the "host" header sent to the debuggee.

Reply:
if add `wss=` or `ws=` query to url for Chrome devtools
`devtools://devtools/bundled/inspector.html?experiments=true&ws=demo:41000`
it will pass `demo` as Host header, differed from js-debug.

I hava another idea, create a new extension to simulate discover service (`/json`) and reverse proxy websocket instead of modify js-debug.

Reply:
> I hava another idea, create a new extension to simulate discover service (/json) and reverse proxy websocket instead of modify js-debug.

Alternately, it could be a config option for `attach` configuration, like `remoteHostHeader`